docs/source/architecture.rst

```diff
@@ -0,0 +1,34 @@
+Architecture
+============
+
+Cloudformation
+--------------
+
+Cloudformation will create the CI/CD components utilizing:
+
+    * **CodePipeline**
+    * **CodeCommit**
+    * **CodeBuild**
+    * **s3**
+    * **CloudWatch**
+    * **Cloudtrail**
+    * **GitHub**
+
+The Cloudformation template has the potential to be separately deployed for various event sources, while the only difference will be the CodeCommit repository that contains the custom configuration data for that event source.
+This allows us to maintain a generic framework that is event source agnostic and relieved of sensitive data while still maintaining configuration revision history.
+
+Serverless
+----------
+
+The Serverless framework deploys the application components that will handle the event source data. This framework creates the following resources:
+
+    * A **custom authorizer** for webhook based APIs that use an Authorization header.
+    * An **API Gateway** for receiving post events using the authorizer.
+    * Custom configuration variables that are populated by the buildspec.yml located in each event source configuration's **CodeCommit** repository.
```

Comment:
Similar to my above comment...To keep it generic we should probably say "...hat are populated by the buildspec.yml located in each event source configuration."

Reply:
I'll resolve this with the use of footnotes to inform.